### PR TITLE
fix: maintainer permission greater than scope

### PR DESCRIPTION
**Pull request by XadillaX**

### CODE REVIEW

**Comment by alsotang** [comment]
Why this pr do the same thing as https://github.com/cnpm/cnpmjs.org/blob/master/middleware/editable.js ?

In my opinion, `publishable.js` is about checking pkg name, and `editable.js` is about checking pkg permission.

Is it any bug occurs here?

**Comment by XadillaX** [comment]
> Why this pr do the same thing as https://github.com/cnpm/cnpmjs.org/blob/master/middleware/editable.js ?
> 
> In my opinion, `publishable.js` is about checking pkg name, and `editable.js` is about checking pkg permission.
> 
> Is it any bug occurs here?

If one is the maintainer of `@foo/bar`, but he's no permission to `@foo/` scope. I think he can `publish`.

But in the previous code, he can't.

**Comment by alsotang** [comment]
Is it `@foo` a valid whitelist scope in the tnpm server? If it isn't, then he should not publish successfully.

**Comment by alsotang** [comment]
> but he's no permission to @foo/ scope 

This line is wrong. It's not his permission. Which scope is valid is based on the whitelist scope config

File: middleware/publishable.js [hunk 1]
```diff
@@ -31,6 +37,22 @@ module.exports = function *publishable(next) {
     return yield next;
   }
 
+  // check the package is already exists and is maintainer
+  var result = yield packageService.authMaintainer(name, this.user.name);
+  if (result.maintainers && result.maintainers.length) {
+    if (result.isMaintainer) {
```
Comment (fengmk2) [inline middleware/publishable.js]: 下面那个测试是测试这个逻辑的？

Comment (XadillaX) [reply middleware/publishable.js]: 是的，先拿走了 `scope` 权限，mock 成 `@xxx`，然后再看看该 maintainer 是否有权限发布。

**Comment by XadillaX** [comment]
> > but he's no permission to @foo/ scope
> 
> This line is wrong. It's not his permission. Which scope is valid is based on the whitelist scope config

If one has no permission to `@foo/`, but the maintainer of `@foo/bar` add him/her to one of the maintainers of `@foo/bar`. How do you think the permission?

**Comment by alsotang** [comment]
If no one has permission to `@foo/`, then no one should modify pkgs under it.

I assume whitelist scope should only add scope, can not remove scope.

**Comment by XadillaX** [comment]
> If no one has permission to `@foo/`, then no one should modify pkgs under it.
> 
> I assume whitelist scope should only add scope, can not remove scope.

Actually the scope may be dynamic by customized `UserService`.

**Comment by fengmk2** [comment]
Release 3.0.0-rc.28

**Pull request merged**